controller/src/beerocks/master/db/network_map.cpp

```diff
@@ -109,20 +115,41 @@ void network_map::send_bml_network_map_message(db &database, Socket *sd,
                 response =
                     message_com::create_vs_message<beerocks_message::cACTION_BML_NW_MAP_RESPONSE>(
                         cmdu_tx, id);
-                beerocks_header                      = message_com::get_beerocks_header(cmdu_tx);
-                beerocks_header->actionhdr()->last() = 0;
-                num_of_nodes = response->node_num(); // prepare for next message
-                num_of_nodes = 0;
-                data_start   = nullptr;
-                size         = 0;
+                if (!response) {
+                    LOG(ERROR) << "create_vs_message() failed ";
+                    return;
+                }
+
+                beerocks_header = message_com::get_beerocks_header(cmdu_tx);
+                if (!beerocks_header) {
+                    LOG(ERROR) << "get_beerocks_header() failed ";
+                    return;
+                }
+
+                auto actionhdr = beerocks_header->actionhdr();
+                if (!actionhdr) {
+                    LOG(ERROR) << "actionhdr() failed ";
+                    return;
+                }
+
+                actionhdr->last() = 0;
+                num_of_nodes      = response->node_num(); // prepare for next message
+                num_of_nodes      = 0;
+                data_start        = nullptr;
+                size              = 0;
             } else {
                 if (!response->alloc_buffer(node_len)) {
                     LOG(ERROR) << "Failed allocating buffer!";
                     return;
                 }
 
-                if (data_start == nullptr) {
-                    data_start = (uint8_t *)response->buffer(0);
+                if (!data_start) {
+                    auto buf = response->buffer(0);
+                    if (!buf) {
+                        LOG(ERROR) << "buffer is nullptr";
+                        return;
+                    }
```

Comment:
Does klocwork really require this? To me, it would be more natural to do:

```
    data_start = (uint8_t *)response->buffer(0);
    if (!data_start) {
```